src/MonthView/phone/CalendarMonth/CalendarMonth.tsx

```diff
@@ -0,0 +1,200 @@
+import moment from "moment"
+import * as React from "react"
+import { LayoutChangeEvent, View } from "react-native"
+import { DateNumber, Month, Task, User } from "../../../shared/model"
+import { styles } from "./CalendarMonth.styles"
+import { DateCalendarBox } from "./DateCalendarBox"
+import { MonthSelection } from "./MonthSelection"
+import { WeekDayHeader } from "./WeekDayHeader"
+
+export interface TaskAndUser {
+    user: User,
+    tasks: Task[]
+}
+
+interface CalendarMonthProps {
+    data?: TaskAndUser[]
+}
+
+interface CalendarMonthState {
+    dateBoxWidth: number,
+    selectedDate?: DateNumber,
+    currentMonth: Month,
+    currentYear: number,
+}
+
+interface BoxToDateNumberMap {
+    [key: number]: DateObject
+}
+
+interface ScheduleToUserMap {
+    [key: number]: User[]
+}
+
+interface DateObject {
+    dateNumber: DateNumber
+    isWeekend: boolean
+    usersBusy: User[]
+    isSelected: boolean
+}
+
+const nBoxToDateMapper = (year: number, month: number,
+                          scheduleToUserMap: ScheduleToUserMap,
+                          selectedDate?: DateNumber): BoxToDateNumberMap => {
+    const date = new Date(year, month, 1)
+    const result: BoxToDateNumberMap = {}
+    let startDayOfTheMonth = date.getDay()
+    while (date.getMonth() === month) {
+        const currentDay = date.getDay()
+        const ms = date.getTime()
+        let usersBusy: User[] = []
+        let isWeekend = false
+        const isSelected = date.getDate() === selectedDate
+        if (currentDay === 0 || currentDay === 6) {
+            isWeekend = true
+        }
+
+        if (scheduleToUserMap[ms]) {
+            usersBusy = scheduleToUserMap[ms]
+        }
+
+        result[startDayOfTheMonth] = {
+            dateNumber: date.getDate() as DateNumber,
+            isWeekend,
+            usersBusy,
+            isSelected
+        }
+
+        startDayOfTheMonth += 1
+        date.setDate(date.getDate() + 1)
+    }
+    return result
+}
+
+const flattenTaskSchedule = (data: TaskAndUser[]): ScheduleToUserMap => {
+    const mapped: ScheduleToUserMap = {}
+    data.forEach((value: TaskAndUser) => {
+        value.tasks.forEach((task: Task) => {
+            const dateOnly = new Date(task.startTime.getFullYear(), task.startTime.getMonth(), task.startTime.getDate())
+            const ms = dateOnly.getTime()
+            let arrayUser: User[] = []
+            if (!mapped[ms]) {
+                arrayUser = []
+            } else {
+                arrayUser = [...mapped[ms]]
+            }
+
+            arrayUser.push(value.user)
+            mapped[ms] = arrayUser
+        })
+    })
+
+    return mapped
+}
+
+export class CalendarMonth extends React.Component<CalendarMonthProps, CalendarMonthState> {
+
+    private boxes = 35
```

Comment:
just an interesting fact, this thing is going to have an extra empty row in February 2026 lol